src/electionguard/api/encrypt_ballot.c

```diff
@@ -113,7 +129,113 @@ void API_EncryptBallot_free(struct register_ballot_message message,
     }
 
     if (tracker_string != NULL)
+    {
         free(tracker_string);
+        tracker_string = NULL;
+    }
+}
+
+bool API_EncryptBallot_soft_delete_file(char *export_path, char *filename)
+{
+    bool ok = true;
+    char *default_prefix = "electionguard_encrypted_ballots-";
+    char *existing_filename = malloc(FILENAME_MAX + 1);
+    if (existing_filename == NULL)
+    {
+        ok = false;
+        return ok;
+    }
+
+    char *soft_delete_filename = malloc(FILENAME_MAX + 1);
+    if (soft_delete_filename == NULL)
+    {
+        ok = false;
+        return ok;
```

Comment:
Potential Memory Leak. By having multiple `return` statements in various of the functions in this file, there is a probability for memory leak as we do not clean memory previously allocated. Potential recommendation that should still be very readable: Allocate all buffers, and then check if there is a an OOM once. NOTE: The same comments applies to multiple functions, but only making the comment once.